src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et3ResponseService.java

```diff
@@ -63,4 +65,26 @@ public void saveEt3ResponseDocument(CaseData caseData, DocumentInfo documentInfo
 
         caseData.getDocumentCollection().add(documentTypeItem);
     }
-}
\ No newline at end of file
+
+    public void saveRelatedDocumentsToDocumentCollection(CaseData caseData) {
+        if (CollectionUtils.isEmpty(caseData.getDocumentCollection())) {
+            caseData.setDocumentCollection(new ArrayList<>());
+        }
+
+        List<DocumentTypeItem> documents = caseData.getDocumentCollection();
+
+        Optional.ofNullable(caseData.getEt3ResponseContestClaimDocument())
+                .orElse(List.of())
+                .stream()
+                .filter(o -> documents.stream().noneMatch(x -> x.getId().equals(o.getId())))
+                .forEach(documents::add);
```

Comment:
My comment still stands for .forEach, you shouldn't really use it for adding values to a collection. It's better to collect and then addAll